diesel/src/pg/types/date_and_time/mod.rs

```diff
@@ -80,6 +80,75 @@ impl PgInterval {
     pub fn from_months(months: i32) -> Self {
         Self::new(0, 0, months)
     }
+
+    fn from_duration(mut seconds: i64, microseconds: i64) -> Self {
```

Comment:
This is very much not the right approach. We cannot assume that `25.hours()` means `1.day() + 1.hour()`. `chrono::Duration` is lossy. We should not try to infer what they meant when they constructed it. We should convert the types with their direct translation:

```
PgInterval {
    microseconds: duration.num_microseconds.expect("Cannot convert a duration larger than 292471 years"),
    ..Default::default(),
}
```